File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Python/s612878490_commented.py
import sys

line = input()

N, M = map(int, line.split())

shortage = []

for _ in range(M):
    A = int(input().split()[0])
    if N > A:
        shortage.append(N - A)

shortage.sort()

ans = sum(shortage[:-1]) if len(shortage) > 1 else 0

print(ans)  # End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/Go/s612878490_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	var N, M int
	if scanner.Scan() {
		line := scanner.Text()
		parts := strings.Split(line, " ")
		N, _ = strconv.Atoi(parts[0])
		M, _ = strconv.Atoi(parts[1])
	}
	shortage := make([]int, M)
	for i := 0; i < M; i++ {
		if scanner.Scan() {
			line := scanner.Text()
			A, _ := strconv.Atoi(line)
			if N > A {
				shortage[i] = N - A
			} else {
				shortage[i] = 0
			}
		}
	}
	sort.Ints(shortage)
	sum := 0
	for i := 0; i < len(shortage)-1; i++ {
		sum += shortage[i]
	}
	fmt.Println(sum)
}

// End of Code
